src/main/java/com/google/sps/data/DatastoreAccess.java

```diff
@@ -0,0 +1,390 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.util.ContextFields;
+import com.google.sps.util.ParameterConstants;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.TimeZone;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class DatastoreAccess implements DataAccess {
+
+  private static boolean seeded = false;
+  private static Random rnd = new Random();
+  private static DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+  private static UserService userService = UserServiceFactory.getUserService();
+
+  public DatastoreAccess() {
+    if (!DatastoreAccess.seeded) {
+      seed_db();
```

Comment:
do you want to seed every time your service comes up? - would it be better to move the seeding logic to a different script that you can run on demand?